src/components/Searcher.js

```diff
@@ -930,22 +930,22 @@ class Searcher extends React.Component<SearcherDefaultProps, SearcherProps, Sear
   /**
    * Add a facet filter to the current search. Will repeat the search
    * if it's already been performed. Note that if a filter for the
-   * same facet name already exists, it will be replaced with the
-   * new one.
+   * same facet name already exists, it will add the new filter
+   * instead of replacing.
    */
   addFacetFilter(facetName: string, bucketLabel: string, filter: string) {
+    const updatedFacetFilters = this.state.facetFilters ? this.state.facetFilters : [];
+    const filterAlreadyExisting = updatedFacetFilters.some((facetFilter: FacetFilter) => {
+      return facetFilter.filter === filter;
+    });
+    if (filterAlreadyExisting) {
```

Comment:
Is this just to handle the case for the tag cloud where the user can potentially click the same value over and over? If so, maybe add a comment there...

Although, I think it would be better to make sure that the values in the tag cloud for already applied filters shouldn't be clickable. You'd need to do this in the `TagCloudFacetContents` component, I think... (I think Dorit mentions something like this in the ticket.)

Reply:
Do you mean that tag cloud buckets that are actually applied as filters should render as `<span>` instead of `<a>`?
If so, logic for `<TagCloud>` would probably need to be changed. As it does not currently have a mechanism to make certain buckets non-clickable.

Reply:
Yes... that's what I mean. They would still show up with the correct size and all, but they:
1. shouldn't be clickable (if changing the <a> to <span> is the best way to do that, that's great)
2. shouldn't do the scaling thing that Nikhil added (where when you hover over them they grow)

It may be that the change for #1 will make #2 happen automatically, depending on how the CSS is set up.

The <TagCloud> component already has a mechanism for making the whole thing not clickable (the `noLink` parameter). You'll need to add a flag like that (I suppose using the same name would make things simpler) to the `TagCloudValue` class (the default value of which should be false). Then you'll also need to modify the TagCloudFacetContents to find out which ones, if any, are already applied.

Does that makes sense?

Reply:
Yeah makes sense. I had the same approach in my mind! You described it far better :)

So, I have added a `noLink` property in the `TagCloudValue` class which will be set from `TagCloudFacetContents` if the search filter is applied for that tag cloud. In `<TagCloud>`, the tag is rendered as a `<span>` if `noLink` is set for that tag.

However, regarding scaling of the tag clouds, it is controlled in the CSS and not in complete control by suit. The scaling CSS property is added on the CSS class `attivio-cloud` which is the class for the parent `<ul>` for all child `<li>` tag values.
So, a solution could be to add a new class `attivio-cloud-noLink` added to `<li>` tags which have `noLink = true` which will prevent the scaling. This would ensure existing behavior stays intact as well...

Reply:
That sounds like a the way to go. You'll be able to prevent both the scaling and the color change that happen now when hovering, while still preserving the font size, etc., so they still display at the right weight according to the count for that bucket.